grub/02_ignition_firstboot

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+exec tail -n +3 $0
+set ignition_firstboot=""
+# Determine if this is a first boot.
+if [ -f "/ignition-firstboot" ]; then
+    set ignition_firstboot="ignition.firstboot=detected"
```

Comment:
so I don't think we actually need the `=detected` part. can we try it just with `ignition.firstboot` by itself?

Reply:
The `=detected` was there for a reason...it relates to the diskless PXE case I think.  @bgilbert would know.

Reply:
I've removed it for now, let me know if we need it

Reply:
In CL the difference is: PXE sets `coreos.first_boot=1`, which runs Ignition.  GRUB sets `coreos.first_boot=detected`, which runs Ignition and then mounts `/boot` to delete the flag file.  If we didn't distinguish between those, PXE would wait forever for the (nonexistent) `/boot` filesystem to show up.

Reply:
in FCOS right now I think we are using `ignition-firstboot-complete.service` to remove the file so I think we don't have that same problem?

Reply:
As of this PR, we still do:

```
ConditionKernelCommandLine=ignition.firstboot
RequiresMountsFor=/boot
```

Dropping the `RequiresMountsFor` would create a race in the other direction, if `/boot` is slow to mount.  It might be possible to condition on PXE rather than on a distinct firstboot cmdline value, if we have a good way to detect PXE.

Note that conditionalization can't happen in `ignition-firstboot-complete.service` because systemd would still unconditionally pull in the unit's dependencies, including `/boot`.  The service will need to be conditionally enabled by a generator instead.

Reply:
not too pretty, but maybe this would work:
`ignition-firstboot-non-pxe.service`:
```
[Unit]
ConditionKernelCommandLine=ignition.firstboot
ConditionPathIsMountPoint=/boot

[Service]
Type=oneshot
ExecStart=/bin/systemctl start ignition-firstboot-non-pxe-cleanup.service
```
`ignition-firstboot-non-pxe-cleanup.service`:
```
[Unit]
RequiresMountsFor=/boot

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/bin/rm /boot/ignition.firstboot
```

Reply:
I've been thinking about this a bit more. I've never run a diskless PXE case (I've run PXE for installs, but not for diskless). I realize in the PXE case why we don't **need** a /boot/ partition (i.e. kernel and initrd are loaded via PXE over network), but why not have the disk image that gets mounted just be the same disk image that we serve everywhere else? i.e. so /boot/ does exist and the file can get deleted without issue (even though the delete would be pointless).

Reply:
discussed a little bit in an open discussion today. i think for now we agreed to leave `ignition.firstboot=detected` out and add it back once we add in PXE support if we still want to implement it the same way it was done in the past for CL

Reply:
@LorbusChris `ExecStart=/bin/systemctl start` is never a good idea, since it hides dependencies from systemd's dependency system.  Also, I believe `ConditionPathIsMountPoint` detects whether the specified directory is a mountpoint at this exact moment, so we'd still have the race.  A systemd generator is the right way to handle conditional unit enablement.

@dustymabe The PXE image would have the same files, but not necessarily the same partition layout.  systemd would need to know not to try to mount a dedicated partition on `/boot`.

After more thought, I'm not sure `RequiresMountsFor` is actually the problem.  It seems as though systemd in the real root should know whether the ESP should be mounted on `/boot`; however, we'll still need a generator to conditionalize that.